src/transformers.ts

```diff
@@ -58,53 +60,43 @@ export default function transform(payload: any, transformers: Transformer[]): an
 
 // dropProperties removes all specified props from the object.
 function dropProperties(payload: any, config: TransformerConfig) {
-  for (const key in config.drop) {
-    if (!config.drop.hasOwnProperty(key)) {
-      continue
-    }
-
-    // If key is empty, it refers to the top-level object.
-    const field = key === '' ? payload : get(payload, key)
-
-    // Can only drop props off of arrays and objects.
-    if (typeof field !== 'object' || field === null) {
-      continue
-    }
-
-    for (const target of config.drop[key]) {
-      delete field[target]
-    }
-  }
+  filterProperties(payload, config.drop, (matchedObj, dropList) => {
+    dropList.forEach(k => delete matchedObj[k])
+  })
 }
 
 // allowProperties ONLY allows the specific targets within the keys. (e.g. "a.foo": ["bar", "baz"]
 // on {a: {foo: {bar: 1, baz: 2}, other: 3}} will not have any drops, as it only looks inside a.foo
 function allowProperties(payload: any, config: TransformerConfig) {
-  for (const key in config.allow) {
-    if (!config.allow.hasOwnProperty(key)) {
-      continue
-    }
-
-    // If key is empty, it refers to the top-level object.
-    const field = key === '' ? payload : get(payload, key)
+  filterProperties(payload, config.allow, (matchedObj, preserveList) => {
+    Object.keys(matchedObj).forEach(key => {
+      if (!preserveList.includes(key)) {
+        delete matchedObj[key]
+      }
+    })
+  })
+}
 
-    // Can only drop props off of arrays and objects.
-    if (typeof field !== 'object' || field === null) {
-      continue
+function filterProperties(payload: any, ruleSet: KeyTarget, filterCb: (matchedObject: any, targets: string[]) => void) {
+  Object.entries(ruleSet).forEach(([key, targets]) => {
+    const filter = (obj: any) => {
+      // Can only act on objects.
+      if (typeof obj !== 'object' || obj === null) {
+        return
+      }
+      
+      filterCb(obj, targets)
     }
 
-    // Execution order fortunately doesn't really matter (e.g. if someone filtered off of foo.bar, then foo.bar.baz)
-    // except for micro-optimization.
-    for (const k in field) {
-      if (!field.hasOwnProperty(k)) {
-        continue
-      }
+    // If key is empty, it refers to the top-level object.
+    const matchedObject = key === '' ? payload : get(payload, key)
 
-      if (config.allow[key].indexOf(k) === -1) {
-        delete field[k]
-      }
+    if(Array.isArray(matchedObject)) {
```

Comment:
space after this if here? dang, no prettier =)

Reply:
```suggestion
    if (Array.isArray(matchedObject)) {
```